src/pages/ProductDetail.vue

```diff
@@ -0,0 +1,74 @@
+<template>
+    <NavBar/>
+    <main class="flex justify-center">
+       
+		<div id="details" class="details p-5 flex mt-28 w-[900px] border sm:flex-col sm:mx-44">
+            <div class="relative mx-5">
+                <div>
+                    <div class="flex">
+                        <SwiperCard :productImages="productList.images" class="h-80 w-96" />                          
+                    </div>
+               </div>
+            </div>
+            <div class="ml-10 sm:ml-0">
+                <p class="text-xl font-bold">{{productList.title}}</p>
+
+                <div>
+                    <p class="text-[12px]"><b>Brand</b> | {{productList.brand}}</p>
+                </div>
+
+                <star-rating :rating="productList.rating" :read-only="true" :increment="0.01" star-size="25"></star-rating>
+
+                <p class="font-bold text-emerald-500 text-xl mt-5">Rs.{{calculateDiscoutedAmount()}}</p>
+
+                <div class="flex">
+                    <p class="text-gray-500 mr-3 line-through text-sm">Rs.{{productList.price}}</p>
+                    <p class="text-sm">-{{productList.discountPercentage}}%</p></div><div class="pt-5">
+                        <button class="bg-orange-500 font-bold text-white mr-5 w-44 h-10">Buy Now</button>
+                        <button class="bg-blue-500 font-bold w-44 h-10 text-white">Add to Cart</button>
+                </div>
+
+                <div>
+                    <p class="font-bold pt-3">Product Details</p>
+                    <p class="text-sm">{{productList.description}}</p>
+                </div>
+            </div>
+        </div>
+	</main>
+</template>
+
+<script>
+import axios from 'axios'
+import NavBar from '../components/NavBar.vue'
+import SwiperCard from '../components/SwiperCard.vue'
+import {useRoute} from 'vue-router';
+import StarRating from 'vue-star-rating'
+export default {
+    name: "ProductDetail",
+    data()
+    {
+        return{
+            productList:[],
+        }
+    },
+     components: {
+        NavBar,
+        StarRating,
+        SwiperCard,
+    },
+    methods:{
+      calculateDiscoutedAmount(){
+          return (this.productList.price-((this.productList.price*this.productList.discountPercentage)/100)).toFixed(2)
+      },
```

Comment:
try looking into mixin so you can use this method in multiple places instead of defining same method in two places

Reply:
done